### PR TITLE
Modify Understanding init containers section

### PR DESCRIPTION
**Pull request by seongpyoHong**
Issue #24318

### CODE REVIEW

File: content/en/docs/concepts/workloads/pods/init-containers.md [hunk 1]
```diff
@@ -28,8 +28,8 @@ Init containers are exactly like regular containers, except:
 * Init containers always run to completion.
 * Each init container must complete successfully before the next one starts.
 
-If a Pod's init container fails, Kubernetes repeatedly restarts the Pod until the init container
-succeeds. However, if the Pod has a `restartPolicy` of Never, Kubernetes does not restart the Pod.
+If a Pod's init container fails, the kubelet repeatedly restarts that init container it container succeeds.
```
Comment (chearmstrong) [inline content/en/docs/concepts/workloads/pods/init-containers.md:31]:
```suggestion
If a Pod's init container fails, the kubelet repeatedly restarts that init container until it succeeds. 
```

The sentence doesn't make sense. Should it not be this?

Comment (seongpyoHong) [reply content/en/docs/concepts/workloads/pods/init-containers.md:31]: Right. The sentence you fixed is correct.
I'll commit the suggestion.

File: content/en/docs/concepts/workloads/pods/init-containers.md [hunk 2]
```diff
@@ -28,8 +28,8 @@ Init containers are exactly like regular containers, except:
 * Init containers always run to completion.
 * Each init container must complete successfully before the next one starts.
 
-If a Pod's init container fails, Kubernetes repeatedly restarts the Pod until the init container
-succeeds. However, if the Pod has a `restartPolicy` of Never, Kubernetes does not restart the Pod.
+If a Pod's init container fails, the kubelet repeatedly restarts that init container until it succeeds.
```
Comment (seongpyoHong) [inline content/en/docs/concepts/workloads/pods/init-containers.md:31]: I changed it.

**Review by sftim — COMMENTED** [review]
/lgtm
/approve

Thanks @seongpyoHong

**Comment by seongpyoHong** [comment]
close #24318

**Pull request merged**